### ISSUE TITLE
 3.8.0 introduced behavior change in `addHttpHeader`

### ISSUE THREAD

**Issue by ychescale9** [apollographql/apollo-kotlin#4833]
### Version

3.8.0

### Summary

https://github.com/apollographql/apollo-kotlin/pull/4754 seems to have introduced a behavior change:

https://github.com/apollographql/apollo-kotlin/blob/d8191287004b98c3468d7f0fdc82caac0fb94793/libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloCall.kt#L66-L68

as the headers are added to the private `additionalHttpHeaders` instead of `httpHeaders`, we can no longer inspect the current `httpHeaders` in an `ApolloCall` and make further changes to the headers based on the current headers:

```kt
// `this` is ApolloCall<T>
val header = httpHeaders?.find { it.name == headerName }
if (header == null) {
    addHttpHeader(headerName, headerValues)
} else {
    val newValue = header.value.plus(",$headerValues")
    httpHeaders = httpHeaders?.replace(headerName, newValue)
}
```

Using `httpHeaders(listOf(HttpHeader(HEADER_KEY, HEADER_VALUE)))` instead as the document suggested works, but the behavior change caught by our tests is a bit unexpected.

**Comment by BoD** [comment]
Hi!

Thanks for reporting this! Indeed, this part of the code has changed in the context of [this issue](https://github.com/apollographql/apollo-kotlin/issues/4753).

Agreed that this is unexpected, and it would make sense for the additional headers to be part of what  `ApolloCall.httpHeaders` returns. I'll take a look at a fix!

**Comment by BoD** [comment]
The fix was released as part of v3.8.1.
